overlord/snapstate/backend/copydata_test.go

```diff
@@ -83,6 +84,7 @@ func (s *copydataSuite) TestCopyData(c *C) {
 }
 
 func (s *copydataSuite) testCopyData(c *C, snapDir string, opts *dirs.SnapDirOptions) {
+	dirs.SetSnapHomeDirs(filepath.Join(s.tempdir, "home"))
```

Comment:
all the setting of SetSnapHomeDirs in the tests in this package need to be undone capturing the old value before rootdir is change and putting it back, also some of these might not be needed if we make SnapHomeDirs respect dirs.GlobalRootDir

Reply:
Thank you for the review! I completely agree, after doing some changes to `SetSnapHomeDirs` I even discovered  that on some tests s.tempdir was having different values for unit tests in the same package, which was causing issues and leading to `SetSnapHomeDirs` generating paths that look like the following `[]string{s.tempdir1/s.tempdir2/homedirs}`. I went through all the instances of s.tempdir used in the unit tests and removed them when they were not needed or replaced them by `dirs.GlobalRootDir` if it was used to generate an expected path for `c.Check`. 
In most unit tests `s.tempdir` should not be used except in `SetUpTest` to initialize the root directory.